src/components/dialogs/PinCodeDialog.tsx

```diff
@@ -0,0 +1,140 @@
+import InputMask from 'react-input-mask'
+import React, {
+  ChangeEvent,
+  KeyboardEvent,
+  useEffect,
+  useRef,
+  useState
+} from 'react'
+
+import TextField from 'cozy-ui/transpiled/react/MuiCozyTheme/TextField'
+import Typography from 'cozy-ui/transpiled/react/Typography'
+import { Button } from 'cozy-ui/transpiled/react/Button'
+import { Dialog } from 'cozy-ui/transpiled/react/CozyDialogs'
+import { useI18n } from 'cozy-ui/transpiled/react/I18n'
+
+interface PinCodeDialogProps {
+  setModalVisible: React.Dispatch<React.SetStateAction<boolean>>
+  setPinCode: (pinCode: string) => void
+}
+
+const validatePinCode = (pinCode: string): boolean =>
+  pinCode.replace(/_/g, '').length === 4
+
+export const PinCodeDialog = ({
+  setPinCode,
+  setModalVisible
+}: PinCodeDialogProps): JSX.Element => {
+  const [confirmValue, setConfirmValue] = useState('')
+  const [error, setError] = useState('')
+  const [flowStep, setFlowStep] = useState<'initial' | 'confirm'>('initial')
+  const [initialValue, setInitialValue] = useState('')
+  const onCancel = (): void => setModalVisible(false)
+  const inputRef = useRef<HTMLInputElement>()
+  const { t } = useI18n()
+
+  const handleChange = (
+    event: ChangeEvent<HTMLInputElement> | KeyboardEvent
+  ): void => {
+    if ('value' in event.target) {
+      flowStep === 'initial'
+        ? setInitialValue(event.target.value)
+        : setConfirmValue(event.target.value)
+    }
+
+    if ('key' in event && flowStep === 'initial') {
+      validatePinCode(initialValue) && setFlowStep('confirm')
+      inputRef.current?.blur()
+    }
+
+    if ('key' in event && flowStep === 'confirm')
+      validatePinCode(initialValue) && trySetPinCode()
+  }
+
+  const trySetPinCode = (): void => {
+    initialValue !== confirmValue
+      ? setError(t('LockScreenView.confirm_pin_error'))
+      : setPinCode(initialValue)
+  }
+
+  useEffect(() => {
+    flowStep === 'confirm' && inputRef.current?.focus()
+  }, [flowStep])
+
+  return (
+    <Dialog
+      open
+      onClose={onCancel}
+      size="small"
+      title={t(
+        `LockScreenView.${
+          flowStep === 'initial' ? 'choose' : 'confirm'
+        }_pin_title`
+      )}
+      actions={
+        <>
+          <Button
+            onClick={onCancel}
+            theme="secondary"
+            label={t(
+              `LockScreenView.${
+                flowStep === 'initial' ? 'choose' : 'confirm'
+              }_pin_cancel`
+            )}
+          />
+          <Button
+            onClick={(): void =>
+              flowStep === 'initial' ? setFlowStep('confirm') : trySetPinCode()
+            }
+            label={t(
+              `LockScreenView.${
+                flowStep === 'initial' ? 'choose_pin_next' : 'confirm_pin_next'
+              }`
+            )}
+            disabled={
+              flowStep === 'initial'
+                ? initialValue.replace(/_/g, '').length < 4
+                : confirmValue.replace(/_/g, '').length < 4
```

Comment:
Maybe `validatePinCode` can be used here (even if it has a `===` instead of a `<`)?

Reply:
Definitely, seems like I forgot to use it in fact